### PR TITLE
 Juniper Upgrade - Changed inheritance for OrganizationMemberBackend

### PR DESCRIPTION
**Pull request by johnbaldwin**
This commit changes the base model to AllowAllUsersModelBackend from
ModelBackend for which Organizations.backends.OrganizationMemberBackend
inherits. The purpose of this is to address an architectural issue with how
Open edX handles authentication and authorization on email vericification.

See the source code in this commit for more details
https://appsembler.atlassian.net/browse/RED-1864

### CODE REVIEW

File: organizations/backends.py [hunk 1]
```diff
@@ -30,10 +90,12 @@ def authenticate(self, *args, **kwargs):  # pylint: disable=arguments-differ
                 return user
             if user.is_superuser:
                 return user
+
+        # we expect to get here for a non is_useruser on the customer site
         return None
 
 
-class OrganizationMemberBackend(ModelBackend):
+class OrganizationMemberBackend(NoActiveCanAuthenticateMixin, ModelBackend):
```
Comment (OmarIthawi) [inline organizations/backends.py]:
```suggestion
class OrganizationMemberBackend(AllowAllUsersModelBackend):
```

File: organizations/backends.py [hunk 2]
```diff
@@ -11,6 +11,66 @@
 )
 
 
+class NoActiveCanAuthenticateMixin:
```
Comment (OmarIthawi) [inline organizations/backends.py]: I've read the documentation the pointers you've posted which lead me to [lms/envs/common.py](https://github.com/appsembler/edx-platform/blob/7960ef1a18657ae0cd2bd55a14d53dfa5365f577/lms/envs/common.py#L780-L784):

```py
AUTHENTICATION_BACKENDS = [
    'rules.permissions.ObjectPermissionBackend',
    'openedx.core.djangoapps.oauth_dispatch.dot_overrides.backends.EdxRateLimitedAllowAllUsersModelBackend',
    'bridgekeeper.backends.RulePermissionBackend',
]
```

Looking at `EdxRateLimitedAllowAllUsersModelBackend` it turns out it extends `AllowAllUsersModelBackend` which does exactly what `NoActiveCanAuthenticateMixin` does.

```py
# Standard Django backend.
# see https://docs.djangoproject.com/en/3.1/ref/contrib/auth/#django.contrib.auth.backends.AllowAllUsersModelBackend
class AllowAllUsersModelBackend(ModelBackend):
    def user_can_authenticate(self, user):
        return True
```

**Review by OmarIthawi — COMMENTED** [review]
First, thank you for debugging through this very difficult issue! This is a very challenging part of Open edX! I read the comments and reviewed the PR and I think this is the way to go.

I have one suggestion and it should be good to be merged:

 - Import `from django.contrib.auth.backends import AllowAllUsersModelBackend` instead of `ModelBackend`
 - `AllowAllUsersModelBackend` should be used instead of `NoActiveCanAuthenticateMixin`

**Comment by johnbaldwin** [comment]
@OmarIthawi Great find and suggestion! Implemented and also updated documentation. Please review and approve if you don't see anything else

**Review by OmarIthawi — APPROVED** [review]
Thank you @johnbaldwin! Look great!

**Pull request merged**